software/o_c_REV/APP_ENVGEN.ino

```diff
@@ -192,9 +198,19 @@ public:
 
     OC::DigitalInput trigger_input = get_trigger_input();
     uint8_t gate_state = 0;
-    if (triggers & DIGITAL_INPUT_MASK(trigger_input))
-      gate_state |= peaks::CONTROL_GATE_RISING;
 
+    bool triggered = triggered_;
+    if (triggers & DIGITAL_INPUT_MASK(trigger_input)) {
+      sinceTrigger_ = 0;
+      triggered=true;
+    }
+    uint16_t trigger_delay = get_trigger_delay();
+    if ((!trigger_delay && triggered) || (triggered && (sinceTrigger_ >= trigger_delay))) {
```

Comment:
> On 4 Apr 2016, at 10:53 PM, Patrick Dowling <notifications@github.com> wrote:
> 
> In software/o_c_REV/APP_ENVGEN.ino <https://github.com/patrickdowling/O_C/pull/18#discussion_r58370452>:
> 
> >  
> > +    bool triggered = triggered_;
> > +    if (triggers & DIGITAL_INPUT_MASK(trigger_input)) {
> > +      sinceTrigger_ = 0;
> > +      triggered=true;
> > +    }
> > +    uint16_t trigger_delay = get_trigger_delay();
> > +    if ((!trigger_delay && triggered) || (triggered && (sinceTrigger_ >= trigger_delay))) {
> How does this work, sinceTrigger_ never gets updated?
> 
sinceTrigger_ is an instance of the ellapsedMillis class - see https://www.pjrc.com/teensy/td_timing_elaspedMillis.html
> Also, doesn't it lose additional triggers that happen while the delay is active? That's not what I'd expect.
> 
> 

I thought about that, and decided that new triggers should reset the delay timer. So maybe it needs another setting to determine that behaviour.

Tim C